### PR TITLE
Battleship

### PR DESCRIPTION
**Pull request by men32z**
- Player, Gameboard Ship and Position factories (Es6 Classes)
- Tests 
- Dom and logic files
- Basic UI
- Drag and drop logic.

### CODE REVIEW

File: src/js/dom.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  const turnMessage = (player) => {
+    message(`${player.ai ? 'Computer\'s' : 'your'} turn`);
+  };
+
+  return {
+    preRender, render, setController, winnerMessage, turnMessage,
+  };
+})();
+
+export default dom;
```
Comment (MauricioRobayo) [inline src/js/dom.js]: I am curious about the decision you took to use an IIFE here with the revealing module pattern. You are using ES6 modules, no need to protect the global scope as this is going to be on its own module, Webpack will take care of that:

```js
let controller;
const setController = contro => {
  controller = contro;
};
// ...
// ...
// ...
const turnMessage = player => {
  message(`${player.ai ? "Computer's" : "your"} turn`);
};

export default {
  preRender,
  render,
  setController,
  winnerMessage,
  turnMessage
};
```

Or even more proper: 

```js
// ...

export {
  preRender,
  render,
  setController,
  winnerMessage,
  turnMessage
};
```

But then you will need to modify your `import`.

File: src/js/gameLogic.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        setTimeout(() => {
+          handleClick(false, shoot);
+        }, 2000);
+      }
+    }
+  };
+
+  return { setUp, handleClick };
+})();
+export default gameLogic;
```
Comment (MauricioRobayo) [inline src/js/gameLogic.js:66]: Again, not sure why you decided to use this pattern here 😕.

File: src/js/initialDom.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        document.getElementById(`position-${i}`).addEventListener('drop', event => dropping(event));
+      }
+    }
+  };
+
+
+  return { preRender, render };
+})();
+
+export default dom;
```
Comment (MauricioRobayo) [inline src/js/initialDom.js:70]: 🤔

**Review by MauricioRobayo — CHANGES_REQUESTED** [review]
Hi!

Congratulations!

The code is clean and organized, the game works as expected, you have tests 🥇, all requirements are met, but... there is one important [game  rule](https://en.wikipedia.org/wiki/Battleship_(game)#Description) that is missing:

> When all of the squares of a ship have been hit, the ship's owner announces the sinking of the Carrier, Submarine, Cruiser/Destroyer/Patrol Boat, or the titular Battleship.

* [ ] Provide feedback once a ship from the enemy is sunk.

Status: __Request changes__.

Optional: Comments on the code are just for you to give it a though.

File: src/js/dom.js (hunk 1 above)
Comment (jcromerohdz) [reply src/js/dom.js]: Thanks for the tip we didn't about it, Thanks again @MauricioRobayo

**Review by azdrenymeri — APPROVED** [review]
### Hi @men32z and @jcromerohdz 
#### Provide feedback once a ship from the enemy is sunk. :heavy_check_mark: 
#### No further requirements, you can merge it :+1:

**Pull request merged**